drivers/lora/sx126x.c

```diff
@@ -500,7 +507,6 @@ static int sx126x_lora_init(const struct device *dev)
 	dev_data.spi_cfg.frequency = DT_INST_PROP(0, spi_max_frequency);
 	dev_data.spi_cfg.slave = DT_INST_REG_ADDR(0);
 
-	dev_data.mode = MODE_SLEEP;
```

Comment:
`sx12xx_init` calls `Radio.Init` which results in `dev_data.mode` being setup correctly via `SX126xInit`.
While we could manually set `dev_data.mode` after the init call, there is no guarantee that we are actually setting the value to the correct mode given the calls `Radio.Init` makes.

Unfortunately it isn't obvious that this is the case because of the call structure, or even that it is initialized at all.
Perhaps we just need a comment above `sx12xx_init` stating that it correctly sets up the variable?

Reply:
As justification for why this shouldn't be manually set, the current value is actually incorrect for this modem until #31568 is merged.